fv3core/decorators.py

```diff
@@ -260,14 +277,20 @@ def decorated(
         domain: Index3D,
         **kwargs,
     ):
-        stencil_config = global_config.get_stencil_config()
-        if isinstance(origin, Mapping):
-            origin_key: Hashable = tuple(sorted(origin.items(), key=lambda x: x[0]))
-            origin_tuple: Tuple[int, ...] = origin["_all_"]
-        else:
+        try:
+            stencil_config = stencil_config_holder[0]
+        except IndexError:
+            stencil_config = global_config.get_stencil_config()
+            stencil_config_holder.append(stencil_config)
+        try:  # works if origin is a Mapping
+            origin_key: Hashable = tuple(
+                sorted(origin.items(), key=lambda x: x[0])  # type: ignore
+            )
+            origin_tuple: Tuple[int, ...] = origin["_all_"]  # type: ignore
+        except AttributeError:  # assume origin is a tuple
             origin_key = origin
-            origin_tuple = origin
-        key: Hashable = (origin_key, domain, stencil_config)
+            origin_tuple = cast(Index3D, origin)
+        key: Hashable = (origin_key, domain)
```

Comment:
can you be sure to incorporate the spec.grid.rank in the key so 54 rank sequential tests will still work? I assume you will when you update to master.

Reply:
Yeah, I'll keep an eye out for that when I resolve merge conflicts.